### PR TITLE
[PDI-14522] - When sftp server has default user folder, absolute urls don't work

### PR DESCRIPTION
**Pull request by YuryBY**
@mattcasters, @brosander, @mattyb149 take a look, please.
What was done: 
1) USER_DIR_IS_ROOT is set to false by default. 
2) Is added corresponding flag to pentaho-variables.xml in order to user has possibility to change USER_DIR_IS_ROOT apache.vfs property
JUnit tests are also provided.

### CODE REVIEW

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local
3. ant -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  core/src/org/pentaho/di/core/Const.java
  core/src/org/pentaho/di/core/util/EnvUtil.java
  core/test-src/org/pentaho/di/core/util/EnvUtilTest.java
  engine/src/kettle-variables.xml
  engine/test-src/org/pentaho/di/core/util/KettleVariablesListTest.java

Checkstyle violations:
======================
No errors found

License header violations:
==========================
No violations found








Unit test coverage change:
============
org.pentaho.di.core.util.EnvUtil:
 * instructionPercentCoverageChange : +1.09%
 * branchPercentCoverageChange : +0.91%
 * linePercentCoverageChange : +1.02%
 * complexityPercentCoverageChange :**-0.67%**
    
org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta:
 * instructionPercentCoverageChange :**-0.17%**
    
org.pentaho.di.trans.steps.sortedmerge.SortedMergeMeta:
 * instructionPercentCoverageChange :**-0.18%**
    
org.pentaho.di.trans.steps.symmetriccrypto.symmetriccryptotrans.SymmetricCryptoTransMeta:
 * instructionPercentCoverageChange : +0.28%
 * branchPercentCoverageChange : +3.33%
 * complexityPercentCoverageChange : +1.85%
    
org.pentaho.di.core.KettleVariablesList:
 * branchPercentCoverageChange : +12.50%
 * complexityPercentCoverageChange : +11.11%
    
org.pentaho.di.trans.step.StepMeta:
 * instructionPercentCoverageChange :**-0.19%**
 * linePercentCoverageChange :**-0.27%**
    
org.pentaho.di.job.entries.sql.JobEntrySQL:
 * instructionPercentCoverageChange : +1.64%
 * linePercentCoverageChange : +1.97%

**Comment by YuryBY** [comment]
@mattcasters, @brosander, @mattyb149 any updates?

File: core/test-src/org/pentaho/di/core/util/EnvUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+package org.pentaho.di.core.util;
+
+import junit.framework.TestCase;
+import org.pentaho.di.core.Const;
+
+/**
+ * Created by Yury_Bakhmutski on 11/4/2015.
+ */
+public class EnvUtilTest extends TestCase {
```
Comment (mattyb149) [inline core/test-src/org/pentaho/di/core/util/EnvUtilTest.java]: New unit tests should use JUnit 4 (@Test and such) not JUnit 3 (TestCase)

File: core/test-src/org/pentaho/di/core/util/EnvUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.pentaho.di.core.Const;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by Yury_Bakhmutski on 11/4/2015.
+ */
+public class EnvUtilTest {
+
+  @Test
```
Comment (YuryBY) [inline core/test-src/org/pentaho/di/core/util/EnvUtilTest.java]: @mattyb149, annotation is added

File: core/test-src/org/pentaho/di/core/util/EnvUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.core.util;
+
+import org.junit.Test;
+import org.pentaho.di.core.Const;
+
+import static org.junit.Assert.*;
```
Comment (mattyb149) [inline core/test-src/org/pentaho/di/core/util/EnvUtilTest.java]: Sorry to be picky, but the coding standards say no star imports :-/  My IDE always puts that in and I have to remember to delete it and import them individually...

File: core/test-src/org/pentaho/di/core/util/EnvUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.core.util;
+
+import org.junit.Test;
+import org.pentaho.di.core.Const;
+
+import static org.junit.Assert.assertEquals;
```
Comment (YuryBY) [inline core/test-src/org/pentaho/di/core/util/EnvUtilTest.java]: @mattyb149, imports are changed (also here: https://github.com/pentaho/pentaho-kettle/pull/1905/files#diff-b698b75195b6f56ae9da7ec21fad5a52R29)

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local
3. ant -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  core/src/org/pentaho/di/core/Const.java
  core/src/org/pentaho/di/core/util/EnvUtil.java
  core/test-src/org/pentaho/di/core/util/EnvUtilTest.java
  engine/src/kettle-variables.xml
  engine/test-src/org/pentaho/di/core/util/KettleVariablesListTest.java

Checkstyle violations:
======================
No errors found

License header violations:
==========================
No violations found








Unit test coverage change:
============
org.pentaho.di.core.extension.ExtensionPointMap.new PluginTypeListener() {...}:
 * instructionPercentCoverageChange : +20.69%
 * linePercentCoverageChange : +25.00%
 * complexityPercentCoverageChange : +25.00%
 * methodPercentCoverageChange : +25.00%
    
org.pentaho.di.core.util.EnvUtil:
 * instructionPercentCoverageChange : +1.09%
 * branchPercentCoverageChange : +0.91%
 * linePercentCoverageChange : +1.02%
 * complexityPercentCoverageChange :**-0.67%**
    
org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta:
 * instructionPercentCoverageChange :**-0.17%**
    
org.pentaho.di.trans.steps.excelwriter.ExcelWriterStepMeta:
 * instructionPercentCoverageChange : +0.08%
    
org.pentaho.di.core.KettleVariablesList:
 * branchPercentCoverageChange : +12.50%
 * complexityPercentCoverageChange : +11.11%
    
org.pentaho.di.trans.steps.symmetriccrypto.symmetriccryptotrans.SymmetricCryptoTransMeta:
 * instructionPercentCoverageChange : +0.28%
 * branchPercentCoverageChange : +3.33%
 * complexityPercentCoverageChange : +1.85%
    
org.pentaho.di.trans.step.StepMeta:
 * instructionPercentCoverageChange :**-0.19%**
 * linePercentCoverageChange :**-0.27%**

**Comment by YuryBY** [comment]
@mattcasters, @brosander, @mattyb149, any updates?

**Pull request merged**